lib/timers.js

```diff
@@ -376,66 +406,41 @@ const clearTimeout = exports.clearTimeout = function(timer) {
 };
 
 
-exports.setInterval = function(callback, repeat) {
+exports.setInterval = function(callback, repeat, arg1, arg2, arg3) {
   if (typeof callback !== 'function') {
     throw new TypeError('"callback" argument must be a function');
   }
 
-  repeat *= 1; // coalesce to number or NaN
+  var len = arguments.length;
+  var args;
+  if (len === 3) {
+    args = [arg1];
+  } else if (len === 4) {
+    args = [arg1, arg2];
+  } else if (len > 4) {
+    args = [arg1, arg2, arg3];
+    for (var i = 5; i < arguments.length; i++)
+      // extend array dynamically, makes .apply run much faster in v6.0.0
+      args[i - 2] = arguments[i];
+  }
+
+  return createRepeatTimeout(callback, repeat, args);
+};
 
-  if (!(repeat >= 1 && repeat <= TIMEOUT_MAX)) {
+function createRepeatTimeout(callback, repeat, args) {
```

Comment:
IMHO, there's no repetead call, you add a short flow path (and creates a mini and fake cyclomatic complexity), function is in the same script and right under the function (so you just added brackets and more names to read). Function call is slower than direct structured code (and even minimal performance gains, in this call stack queen, is important).

I would not do this separation.